core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java

```diff
@@ -448,7 +449,19 @@ public TimestampTZColumn parseTimestampTZColumn(ResultSet resultSet, String colu
   }
 
   @Override
-  public Map<String, String> getConnectionProperties() {
+  public Map<String, String> getConnectionProperties(JdbcConfig config) {
+    if (config.getDatabaseConfig().getScanFetchSize() == Integer.MIN_VALUE) {
+      // If the scan fetch size is set to Integer.MIN_VALUE, use the streaming mode.
+      return Collections.emptyMap();
+    }
+
+    // Otherwise, use the cursor fetch mode.
     return Collections.singletonMap("useCursorFetch", "true");
   }
+
+  @Override
+  public void setConnectionToReadOnly(Connection connection, boolean readOnly) throws SQLException {
+    // Observed performance degradation when using read-only connections in MySQL. So we do not
+    // set the read-only mode for MySQL connections.
+  }
```

Comment:
We don't call `Connection.setReadOnly(true)` for MySQL.

Reply:
Right, we don't know the exact reason. We just observed performance degradation when we call `Connection.setReadOnly(true)` with MySQL.